### PR TITLE
feat(core/figures): anchor link texts to be shortened

### PR DESCRIPTION
**Pull request by himanish-star**
fixes: #879

### CODE REVIEW

File: src/core/figures.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ export function run(conf, doc, cb) {
     id = id.substring(1);
     if (figMap[id]) {
       $a.addClass("fig-ref");
-      if ($a.html() === "") $a.append(figMap[id].clone());
+      if ($a.html() === "") $a.append(figMap[id].slice(0, 2).clone());
```
Comment (marcoscaceres) [inline src/core/figures.js]: To make this a bit more legible... 

```JS
if ($a.html() === "") {
  const $shortFigDescriptor = figMap[id].slice(0, 2).clone();
  $a.append(shortFigDescriptor);
}
```

**Comment by himanish-star** [comment]
ah ok

**Comment by himanish-star** [comment]
Now: 
image

Before:
image

**Comment by marcoscaceres** [comment]
Looks good. For the `<a>`, we should add a title attribute containing the text we are now taking out. Then users can hover over “figure x” to see what it’s referring to exactly.

**Comment by r12a** [comment]
This all looks very good. Thank you very much!

One question. Not having looked closely at the code, I hope the text 'Figure 1' will not appear in documents in other languages. For example, we have https://w3c.github.io/jlreq/ja/ where 'Figure 1' would be '図1'. I believe we have some localisation code in respec to handle that, and it should work here too.

**Comment by himanish-star** [comment]
Sure thanks @marcoscaceres and @r12a for the review

**Comment by himanish-star** [comment]
image

@r12a, this is ok now?

**Comment by r12a** [comment]
Looks good, though it would be easier to spot details if you could show a picture of the jlreq doc at https://w3c.github.io/jlreq/ja/#kanji_hiragana_and_katakana. There is currently a problem in that references in the text to a figure seem to have a space added after. See the blue highlight in this picture:

screen shot 2018-03-28 at 08 07 37

There should be no spaces in the Japanese text. I'm not sure where this comes from: whether it's from the localization code or from the figure link insertion.

**Comment by r12a** [comment]
(oops, wrong picture, i'm going to edit in place).

**Comment by himanish-star** [comment]
Yeah sure

**Comment by himanish-star** [comment]
image
Does this look right now @r12a?

**Comment by himanish-star** [comment]
Also this (caption)
image

**Comment by himanish-star** [comment]
@marcoscaceres, I have written down the tests too for this! :smile:

**Comment by r12a** [comment]
>  Does this look right now @r12a?

I think it does indeed look correct now. Thanks.

**Comment by himanish-star** [comment]
> I think it does indeed look correct now. Thanks.

Oh, that's nice. I have also set the title attribute of the anchor element to the figure caption

**Comment by r12a** [comment]
> I have also set the title attribute of the anchor element to the figure caption

That's a nice feature.

File: src/core/figures.js [hunk 2]
```diff
@@ -76,7 +76,12 @@ export function run(conf, doc, cb) {
     id = id.substring(1);
     if (figMap[id]) {
       $a.addClass("fig-ref");
-      if ($a.html() === "") $a.append(figMap[id].clone());
+      if ($a.html() === "") {
+        const $shortFigDescriptor = figMap[id].slice(0, 2).clone();
+        const longFigDescriptor = figMap[id].slice(2).clone().text();
+        $a.attr("title", longFigDescriptor.trim());
```
Comment (marcoscaceres) [inline src/core/figures.js]: Small edge case here, but we should only set `title`, if the `title` attribute does not exist.

File: tests/spec/core/figures-spec.js [hunk 3]
```diff
@@ -22,6 +22,42 @@ describe("Core - Figures", function() {
     expect(captions.item(0).textContent).toEqual("Figure 1 PREFIG");
     expect(captions.item(1).textContent).toEqual("Figure 2 IMGTIT");
   });
+
+  it("creates autolinks from the anchor to the figure", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body:
+      makeDefaultBody() +
```
Comment (marcoscaceres) [inline tests/spec/core/figures-spec.js]: Nit: use multiline template strings, instead of concatenation with "+" ... makes code more easy to change later.

Comment (himanish-star) [reply tests/spec/core/figures-spec.js]: ok

File: src/core/figures.js (hunk 2 above)
Comment (himanish-star) [reply src/core/figures.js]: Oops! Didn't think of that!

File: tests/spec/core/figures-spec.js [hunk 4]
```diff
@@ -22,6 +22,53 @@ describe("Core - Figures", function() {
...
+    const ops = {
+      config: makeBasicConfig(),
+      htmlAttrs: {
+        lang: "ja",
+      },
+      body:
+      makeDefaultBody() +
+      `<figure id='fig'> <img src='img' alt=''>
+        <figcaption>漢字と仮名のサイズの示し方</figcaption>
+       </figure>" +
```
Comment (marcoscaceres) [inline tests/spec/core/figures-spec.js]: nit: " +

**Comment by marcoscaceres** [comment]
+ respec@20.4.0 

Great work @himanish-star!
